kotest-property/src/commonMain/kotlin/io/kotest/property/resolution/resolve.kt

```diff
@@ -0,0 +1,38 @@
+package io.kotest.property.resolution
+
+import io.kotest.property.Arb
+import kotlin.reflect.KType
+import kotlin.reflect.typeOf
+
+/**
+ * Resolves a default [Arb] for reified type [A].
+ * See [resolve].
+ */
+@Suppress("UNCHECKED_CAST")
+inline fun <reified A> Arb.Companion.default(): Arb<A> = resolve(typeOf<A>()) as Arb<A>
```

Comment:
I'm wondering if there's some way to use an `ArbResolver` as a context receiver here, to allow users to override Arb resolution.

```suggestion
context(ArbResolver)
inline fun <reified A> Arb.Companion.default(): Arb<A> = resolve(typeOf<A>()) as Arb<A> // should use ArbResolver#resolve
```

Ideally the method below (which could be considered a composite `ArbResolver` as well) would be added to the context by default.

Reply:
I don't want to introduce context receivers to Kotest as that will move us to 1.8 as the minimum version (if context receivers are stable by then). We've been bitten before by experimental stuff so not yet.